[CI] MixedClusterClientYamlTestSuiteIT test {p0=index/100_field_name_length_limit/Test field name length limit synthetic source} failing

**Build Scans:**
- [elasticsearch-intake #41259 / 9.4.3_bwc-snapshots](https://gradle-enterprise.elastic.co/s/63c7ohjycc4us)
- [elasticsearch-pull-request #153328 / 9.3.6_part_6_bwc-snapshots](https://gradle-enterprise.elastic.co/s/qikpdoakisvpo)
- [elasticsearch-pull-request #153079 / 9.3.6_part_6_bwc-snapshots](https://gradle-enterprise.elastic.co/s/yf6t3yoz6phok)

**Reproduction Line:**
```
./gradlew ":qa:mixed-cluster:v9.4.3#mixedClusterTest" -Dtests.class="org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT" -Dtests.method="test {p0=index/100_field_name_length_limit/Test field name length limit synthetic source}" -Dtests.seed=61E75EE1B19E1526 -Dtests.bwc=true -Dtests.locale=khq -Dtests.timezone=Asia/Choibalsan -Druntime.java=26
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('test%20%7Bp0%3Dindex%2F100_field_name_length_limit%2FTest%20field%20name%20length%20limit%20synthetic%20source%7D'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: java.lang.AssertionError: field [test_index.mappings.properties.foo.properties.field_name_not_ignored.type] is null
```

**Issue Reasons:**
- [main] 1 consecutive failures in test test {p0=index/100_field_name_length_limit/Test field name length limit synthetic source}
- [main] 1 consecutive failures in step 9.4.3_bwc-snapshots
- [main] 1 consecutive failures in pipeline elasticsearch-intake
- [main] 3 failures in test test {p0=index/100_field_name_length_limit/Test field name length limit synthetic source} (0.5% fail rate in 592 executions)
- [main] 2 failures in step 9.3.6_part_6_bwc-snapshots (1.0% fail rate in 198 executions)
- [main] 2 failures in pipeline elasticsearch-pull-request (1.0% fail rate in 198 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -376,9 +376,6 @@ tests:
 - class: org.elasticsearch.xpack.stateless.StatelessHollowIndexShardsIT
   method: testUnhollowOnUpdates
   issue: https://github.com/elastic/elasticsearch/issues/151100
-- class: org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT
-  method: test {p0=index/100_field_name_length_limit/Test field name length limit synthetic source}
-  issue: https://github.com/elastic/elasticsearch/issues/151117
 - class: org.elasticsearch.xpack.esql.ccq.MultiClusterSpecChangePointIT
   method: test {csv-spec:change_point.values null column}
   issue: https://github.com/elastic/elasticsearch/issues/151132

